packages/ui/src/components/organisms/DynamicForm/components/RSJVInputAdaters/DateInputAdater/DateInputAdater.tsx

```diff
@@ -1,17 +1,47 @@
-import { DatePickerInput } from '@/components/molecules';
+import { DatePickerChangeEvent, DatePickerInput } from '@/components/molecules';
 import { RJSFInputAdapter } from '@/components/organisms/DynamicForm/components/RSJVInputAdaters/types';
 import { useCallback } from 'react';
 
-export const DateInputAdater: RJSFInputAdapter = ({ id, formData, disabled, onBlur, onChange }) => {
+export const isDateValid = (dateString: string): boolean => {
+  const date = new Date(dateString);
+  if (date.getFullYear() < 1000) return false;
+
+  return true;
+};
+
+export const DateInputAdater: RJSFInputAdapter<string | null> = ({
+  id,
+  formData,
+  disabled,
+  onBlur,
+  onChange,
+}) => {
   const handleBlur = useCallback(() => {
     // @ts-ignore
     onBlur && onBlur(id, formData);
   }, [id, onBlur, formData]);
 
+  const handleChange = useCallback(
+    (event: DatePickerChangeEvent) => {
+      const dateValue: string | null = event.target.value;
```

Comment:
Note explicit typing over leaning on inference tends to do more harm than good in application code in the macro. This line does not seem to justify the type.
```suggestion
      const dateValue = event.target.value;
      
```